presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkMutableRow.java

```diff
@@ -109,4 +111,17 @@ private static RuntimeException serializationNotSupportedException()
         // During shuffle rows are always serialized with PrestoSparkShuffleSerializer.
         return new UnsupportedOperationException("PrestoSparkUnsafeRow is not expected to be serialized with Kryo or standard Java serialization");
     }
+
+    @Override
+    public long getRowCount()
+    {
+        return 1;
+    }
+
+    @Override
+    public long getSize()
+    {
+        checkState(buffer != null, "buffer is expected to be not null");
+        return buffer.remaining();
```

Comment:
Curious why return `buffer.remaining()` (as compared with `buffer.size()`?  Will `buffer.remaining()` change during object lifetime? Maybe add a comment ?

Reply:
Yes. Basically we change the pointers as we advance the iterator, but we don't change the buffer itself. The entire buffer contains many rows (~1mb).